packages/dds/tree/src/test/simple-tree/walkSchema.spec.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import {
+	normalizeFieldSchema,
+	SchemaFactoryAlpha,
+	type TreeNodeSchema,
+} from "../../simple-tree/index.js";
+import {
+	walkAllowedTypes,
+	type AnnotatedAllowedSchema,
+	type SchemaVisitor,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../simple-tree/core/index.js";
+
+function makeAnnotated(type: TreeNodeSchema): AnnotatedAllowedSchema {
+	return {
+		metadata: {
+			custom: "test",
+		},
+		type,
+	};
+}
+
+describe("walk schema", () => {
+	let visitedNodes: TreeNodeSchema[];
+	let visitedAllowedTypes: Iterable<AnnotatedAllowedSchema>[];
+
+	const mockVisitor: SchemaVisitor = {
+		node: (schema) => visitedNodes.push(schema),
+		allowedTypes: (types) => visitedAllowedTypes.push(types),
+	};
+
+	const sf = new SchemaFactoryAlpha("walk schema tests");
+
+	beforeEach(() => {
+		visitedNodes = [];
+		visitedAllowedTypes = [];
+	});
+
+	it("calls visitor on single allowed type", () => {
+		const annotated = makeAnnotated(sf.string);
+
+		walkAllowedTypes([annotated], mockVisitor);
+
+		assert.deepEqual(visitedNodes, [annotated.type]);
+		assert.equal(visitedAllowedTypes.length, 1);
+		assert.deepEqual(Array.from(visitedAllowedTypes[0]), [annotated]);
```

Comment:
I feel like this is a more elegant solution that will show more useful diffs if it fails.
```suggestion
		assert.deepEqual(visitedAllowedTypes.map(array.from), [[annotated]]);
```

Generally, one big deep equal ends up being nicer than a few smaller checks